synapse/lib/ast.py

```diff
@@ -720,31 +723,25 @@ async def run(self, runt, genr):
 
         with s_provenance.claim('stormcmd', name=name):
 
-            if runtsafe:
-
-                genr = await pullone(genr)
-
-                argv = await self.kids[1].compute(runt, None)
-                if not await scmd.setArgv(argv):
-                    return
-
-                async for item in scmd.execStormCmd(runt, genr):
-                    yield item
-
-                return
-
-            async def optsgenr():
+            async def genx():
 
                 async for node, path in genr:
-
                     argv = await self.kids[1].compute(runt, path)
                     if not await scmd.setArgv(argv):
                         return
 
                     yield node, path
 
-            scmd = ctor(runt, runtsafe)
-            async for item in scmd.execStormCmd(runt, optsgenr()):
+            # must pull through the genr to get opts set
+            # ( many commands expect self.opts is set at run() )
```

Comment:
This actually may help with some issues we've had with commands *not* having self.opts set. This was an issue with the Macro command.